scripts/buildfire/components/reactions/reactions.js

```
@@ -0,0 +1,1594 @@
+            };
+
+            return index;
+        }
+    }
+
+    class ReactionsTypes {
+        static itemsReactionsGroupName = {};
+        static groups = null;
+
+        static get TAG() {
+            return "$$reactionsGroups";
+        }
+
+        // options = saved to futuer use
+        static getReactionsGroups(options, callback) {
+            if (!callback || typeof callback !== 'function') {
+                return console.error("callback must be a function!");
+            }
+
+            buildfire.appData.get(this.TAG, (err, result) => {
+                if (err) return callback(err)
+
+                if (!result.data || !result.data.groups || !result.data.groups.length) {
+                    console.error('No reaction groups added yet');
+                    this.groups = [];
+                    return callback(null, []);
+                }
+
+                this.groups = result.data.groups;
+                return callback(null, this.groups);
+            });
+        }
+
+        // options ={groupName, itemId}
+        static getReactionsTypes(options, callback) {
+            const { groupName, itemId } = options;
+
+            if (!callback || typeof callback !== 'function') {
+                return console.error("callback must be a function!");
+            }
+
+            if (!this.groups || !this.groups.length) {
+                return callback("No Reactions Group Added yet");
+            }
+
+            if (!itemId) {
+                return callback("missing itemId", null);
+            }
+
+            let group = null;
+            if (groupName) {
+                for (let i = 0; i < this.groups.length; i++) {
+                    if (this.groups[i].name.toLowerCase() === groupName.toLowerCase()) {
+                        group = this.groups[i];
+                        break;
+                    }
+                }
+                if (!group) {
```

should return an err